src/otx/algorithms/common/utils/__init__.py

```diff
@@ -51,4 +53,10 @@
     "read_py_config",
     "get_default_async_reqs_num",
     "is_xpu_available",
+    "is_hpu_available",
+    "cast_bf16_to_fp32",
 ]
+
+
+if is_hpu_available():
+    import habana_frameworks.torch.gpu_migration  # noqa: F401
```

Comment:
Is it safe way to check "is_hpu_available"?. Does it help to improve training?

Reply:
Yes, gpu migration is a convenient way to convert cuda-related components to hpu-related ones.
It can help catch backend components that are too difficult to be discovered.

Reply:
> This part isn't for checking that HPU is available. It just executes gpu_migration if HPU is available.

If it is not checking that HPU is available, then why this function called "is_hpu_available"? Maybe we can add also a check for hpu or rename the function

Reply:
This is not a definition of function. It's just condition branch.

Reply:
Sorry, my misunderstanding

Reply:
L61 is checking if hpu is available. If available, use GPU migration (L62) to convert cuda-related components.

Reply:
Sorry, guys, for some reason I saw "def" instead of "if" :D